### PR TITLE
Added ThemeMaker example program

### PR DESCRIPTION
**Pull request by Akascape**
I added my ThemeMaker program inside example folder directly.
As I described about the program in #264, it can create themes for customtkinter easily.
@TomSchimansky As you mentioned your thoughts in the discussion, I added the load theme option to edit existing themes and also added a reset button that will reset the current colors to null.

### Now the program looks like this:
ThemeMaker

### CODE REVIEW

**Comment by Akascape** [comment]
@TomSchimansky
> That's a good idea. But I think it would be better to have all widgets in one window, and then a dropdown menu at the bottom where you select a widget and then you select the colors for this widget. And the corresponding widget will automatically update to the new colors, so that you see how the colors match each other. Also it would be nice to have the option to load an existing theme file, if you just want to change some values. But you can also make a PR for your current version, I will have a look at it.

I tried to make a window containing all the widgets like that but I am confused on how to configure everything. Instead it would be easy if we can configure the whole theme of the program with `customtkiner.set_default_color_theme("theme")` method just like the `customtkinter.set_appearance_mode(appearance_mode)`. 

But for now this version will at least improve the theme creation process for customtkiner. 
Also mention it in the docs so that users can know how to make themes easily.

File: examples/ThemeMaker.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+from tkinter.colorchooser import askcolor
+from tkinter import filedialog, messagebox
+import json
+
+#Main Window
+WIDTH = 390
+HEIGHT = 350
+customtkinter.set_appearance_mode("Dark")
+customtkinter.set_default_color_theme("blue")
+self=customtkinter.CTk()
```
Comment (TomSchimansky) [inline examples/ThemeMaker.py:12]: Can you structure the program into a class? Don't just use `self` when its not actually inside a class.

File: examples/ThemeMaker.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+self.title("CustomTkinter ThemeMaker")
+self.geometry(f"{WIDTH}x{HEIGHT}")
+self.resizable(width=False, height=False)
+
+null=None #Default is null
```
Comment (TomSchimansky) [inline examples/ThemeMaker.py:17]: There's no need to define a null variable, just use None inside the dict definition or call the variable default_value or something

File: examples/ThemeMaker.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+        "scrollbar_border_spacing": 4
+    }
+}
+
+
+#list of widgets
+Widgets=['Window', 'Frame', 'Button', 'CheckBox', 'Entry', 'Label', 'Text', 'ProgressBar', 'Slider',
+         'Switch', 'Menu + Dropdown', 'Scrollbar']
+
+#List containing menu content
```
Comment (TomSchimansky) [inline examples/ThemeMaker.py:105]: Maybe you can combine these lists into a dictionary like the following:
```python
widget_attributes = {"Winndow": ["window_bg_color"], "Frame": [..., ...], ...}
```

File: examples/ThemeMaker.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+list7=['progressbar','progressbar_progress','progressbar_border']
+list8=['slider','slider_progress','slider_button','slider_button_hover']
+list9=['switch','switch_progress','switch_button','switch_button_hover']
+list10=['optionmenu_button','optionmenu_button_hover','combobox_border','combobox_button_hover',
+        'dropdown_color','dropdown_hover','dropdown_text']
+list11=['scrollbar_button','scrollbar_button_hover']
+
+#Function for changing the values of menu
+def changemenu():
+    if current=='Window':
```
Comment (TomSchimansky) [inline examples/ThemeMaker.py:122]: If you have a dictionary like described above, you don't need all the if else statements, just use `widget_attributes[current]` or something similar

File: examples/ThemeMaker.py [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+def savethis():
+    dialog=customtkinter.CTkInputDialog(master=self, text="Enter your theme name:", title="Save Theme")
+    try:
+        themename=dialog.get_input()+".json"
+        outfile=open(themename,"w")
+        json.dump(json_data, outfile, indent=2)
+        outfile.close()
+        tkinter.messagebox.showinfo("Exported!","Theme exported successfully!")
+    except:
+        pass
```
Comment (TomSchimansky) [inline examples/ThemeMaker.py:241]: Maybe add a tkinter error message

**Review by TomSchimansky — CHANGES_REQUESTED** [review]

**Comment by Akascape** [comment]
@TomSchimansky Ok, I will make all the changes soon...

File: examples/ThemeMaker.py (hunk 5 above)
Comment (Akascape) [reply examples/ThemeMaker.py:241]: It also shows error when we click cancel/close the input dialog, what to do?

File: examples/ThemeMaker.py (hunk 1 above)
Comment (Akascape) [reply examples/ThemeMaker.py:12]: Changed the self to root_tk

File: examples/ThemeMaker.py (hunk 2 above)
Comment (Akascape) [reply examples/ThemeMaker.py:17]: Done

File: examples/ThemeMaker.py (hunk 3 above)
Comment (Akascape) [reply examples/ThemeMaker.py:105]: Done

File: examples/ThemeMaker.py (hunk 4 above)
Comment (Akascape) [reply examples/ThemeMaker.py:122]: Done

**Comment by TheFallen-Cat** [comment]
I was wondering if i could submit a simple Dictionary made with customtkinter as one of the Example Apps

Check the repo here : https://github.com/TheFallen-Cat/VocabiPy

File: examples/ThemeMaker.py (hunk 5 above)
Comment (TomSchimansky) [reply examples/ThemeMaker.py:202]: You need to check if `dialog.get_input()` retunes a string before appending the `.json`.

Comment (Akascape) [reply examples/ThemeMaker.py]: @TomSchimansky Done, I have improve the error handling in both save and load functions, 
I think its ready to merge. Any more changes?

File: examples/ThemeMaker.py (hunk 1 above)
Comment (Akascape) [reply examples/ThemeMaker.py]: @TomSchimansky Done, changed the whole program structure to class

File: examples/ThemeMaker.py (hunk 5 above)
Comment (Akascape) [reply examples/ThemeMaker.py]: I have removed the CtkInputDialog with tkinter savefile dialog. Now its working properly and makes more sense.

**Comment by Akascape** [comment]
@TomSchimansky I have improved the whole code, please review and decide. :)
Need any other changes?

**Pull request closed without merging**